scripts/get_tau_pet.py

```diff
@@ -1,94 +1,121 @@
-tau_df = tau_df.filter(['EXAMDATE','META_TEMPORAL_SUVR'], axis=1)
+    return (baseline_df)
 
-# convert to datetime
-tau_df['EXAMDATE'] = pd.to_datetime(tau_df['EXAMDATE'])
+def get_tau(file):
 
-# sort both dfs by index, and tau df also by date. Also throws an error
-baseline.sort_values(by='RID', inplace=True)
-tau_df.sort_values(by=['RID', 'EXAMDATE'],inplace=True)
+    ''' Load tau data and return df '''
 
-# group subjects in the tau df, look them up in the baseline df, and if they match return the tau
-# value closest to the session date in baseline. Return a list of dfs, one per subject. This seems
-# a silly and expensive way to do this! But, I ran into all sorts of problems with the datetime format
-# and this is the only way I figured it out...
-dfs = []
-for i, grp in tau_df.groupby(level='RID'):
-    for subject, session in zip(baseline.index, baseline.session):
-        if i == subject:
-            df = grp
-            df.set_index('EXAMDATE', inplace=True)
-            tau = pd.DataFrame(df['META_TEMPORAL_SUVR'][df.index[[df.index.get_loc(session, method='nearest')]]])
-            tau['EXAMDATE'] = tau.index
-            tau.index = [subject]
-            tau.rename_axis('RID',inplace=True)
-            dfs.append(tau)
+    # load flortaucipir spreadsheet
+    tau_df = pd.read_csv(file, index_col=0, header=0, low_memory=False)
 
-# concatenate individual tau dfs
-master_tau = pd.concat(dfs)
+    # keep only relevant columns
+    tau_df = tau_df.filter(['EXAMDATE','META_TEMPORAL_SUVR'], axis=1)
 
-# merge tau data into baseline df
-baseline = baseline.join(master_tau)
+    # convert to datetime
+    tau_df['EXAMDATE'] = pd.to_datetime(tau_df['EXAMDATE'])
 
-# create a new df with only tau values that were collected within a 12month window of the baseline visit
-baseline_tau = baseline[(baseline.EXAMDATE > baseline.date_lower) & (baseline.EXAMDATE < baseline.date_upper)]
+    # sort tau df by index and date. Also throws an error
+    tau_df.sort_values(by=['RID', 'EXAMDATE'],inplace=True)
+    
+    return (tau_df)
 
-# save df
-baseline_tau.to_csv('/Users/natashaclarke/Documents/SIMEXP/cwas_neuro_sandbox/adni_cwas/baseline_tau.csv')
\ No newline at end of file
+def match_baseline_tau(baseline_df, tau_df):
+    
+    '''
+    Group subjects in the tau df, look them up in the baseline df, and if they match return the tau
+    value closest to the session date in baseline. Return a list of dfs, one per subject, merge into
+    baseline df, and then keep only those within a 12 month window. This seems a silly and expensive way 
+    to do this! But, I ran into all sorts of problems with the datetime format and this is the only way 
+    I figured it out...
+    '''
+    
+    tau_dfs_list = []
+    for tau_id, group in tau_df.groupby(level='RID'):
+        for baseline_id, session in zip(baseline_df.index, baseline_df.session):
+            if tau_id == baseline_id:
+                participant_df = group
```

Comment:
This is not needed. why not just change L93 to 
`for tau_id, participant_df in tau_df.groupby(level='RID'):`

Reply:
awesome, thank you!!